src/github.com/matrix-org/dendrite/syncapi/sync/userstream.go

```diff
@@ -120,37 +120,40 @@ func (s *UserStream) TimeOfLastNonEmpty() time.Time {
 
 // GetStreamPosition returns last stream position which the UserStream was
 // notified about
-func (s *UserStream) GetStreamPosition() types.StreamPosition {
-	s.lock.Lock()
-	defer s.lock.Unlock()
+func (s *UserStreamListener) GetStreamPosition() types.StreamPosition {
+	s.userStream.lock.Lock()
+	defer s.userStream.lock.Unlock()
 
-	return s.pos
+	return s.userStream.pos
 }
 
 // GetNotifyChannel returns a channel that is closed when there may be an
 // update for the user.
 // sincePos specifies from which point we want to be notified about
```

Comment:
right, thanks for adding the comment at line 138; this makes more sense now. Please could you also add something about it here - "if we're already past that point in the stream, we'll return a closed channel immediately. otherwise we'll return a channel which will be closed once there is an update". Only with better words and formatting